include/cufinufft/impl.h

```diff
@@ -354,6 +354,115 @@ Notes: the type T means either single or double, matching the
     return ier;
 }
 
+template<typename T>
+T calculate_scale_factor(int rank, finufft_spread_opts opts) 
+{
+    // Calculate the additional scalar for multiplying just for spread and interpolate
+    int n = 100;
+    T h = 2.0 / n;
+    T x = -1.0;
+    T sum = 0.0;
+    for(int i = 1; i < n; i++) {
+      x += h;
+      sum += exp(opts.ES_beta * sqrt(1.0 - x * x));
+    }
+    sum += 1.0;
+    sum *= h;
+    sum *= sqrt(1.0 / opts.ES_c);
+    T scale = sum;
+    if (rank > 1) { scale *= sum; }
```

Comment:
``rank`` would need to be documented. This all suggests that this code should reside outside of (cu)FINUFFT, since it's not part of the NUFFT task.